### PR TITLE
Add UT for query auth check

### PR DESCRIPTION
**Pull request by JackieTien97**

### CODE REVIEW

File: iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/AuthTest.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+      fail(e.getMessage());
+    }
+
+    // use database
+    String databaseTest2 = "test2";
+    // user `root` always succeed
+    Mockito.doNothing().when(authChecker).checkDatabaseVisibility(eq(userRoot), any());
+    // user `user1` can't see DATABASE test1
+    errorMsg = String.format("%s has no privileges on DATABASE %s", user1, databaseTest1);
+    Mockito.doThrow(new AccessDeniedException(errorMsg))
```
Comment (ColinLeeo) [inline iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/AuthTest.java:160]: Returning true or false might be a better option？

File: iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/AuthTest.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    Mockito.doNothing().when(authChecker).checkDatabasePrivilege(eq(userRoot), any(), any());
+    // user `user1` don't hava test1's CREATE privilege
+    String errorMsg =
+        String.format(
+            "%s doesn't have %s privilege on DATABASE %s",
+            user1, TableModelPrivilege.CREATE, databaseTest1);
+    Mockito.doThrow(new AccessDeniedException(errorMsg))
+        .when(authChecker)
+        .checkDatabasePrivilege(eq(user1), eq(databaseTest1), eq(TableModelPrivilege.CREATE));
+    // user `user2` hava test1's CREATE privilege
```
Comment (ColinLeeo) [inline iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/AuthTest.java:125]: have

**Review by ColinLeeo — APPROVED** [review]
LGTM

File: iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/AuthTest.java (hunk 1 above)
Comment (JackieTien97) [reply iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/AuthTest.java:160]: We need to throw AccessDeniedException to user while users do `use XXXX` operation.
I want AccessDeniedException thrown source is one place(that is where the implementation class of ITableAuthChecker which you will implement in the future). In such way, you can control the unified error msg format of AccessDeniedException.

**Pull request merged**